exchanges/huobi/huobi_types.go

```diff
@@ -259,3 +263,73 @@ var (
 	TimeIntervalMohth          = TimeInterval("1mon")
 	TimeIntervalYear           = TimeInterval("1year")
 )
+
+// WsRequest defines a request data structure
+type WsRequest struct {
+	Topic             string `json:"req,omitempty"`
+	Subscribe         string `json:"sub,omitempty"`
+	Unsubscribe       string `json:"unsub,omitempty"`
+	ClientGeneratedID string `json:"id,omitempty"`
+}
+
+// WsResponse defines a response from the websocket connection when there
+// is an error
+type WsResponse struct {
+	TS           int64  `json:"ts"`
+	Status       string `json:"status"`
+	ErrorCode    string `json:"err-code"`
+	ErrorMessage string `json:"err-msg"`
+	Ping         int64  `json:"ping"`
+	Channel      string `json:"ch"`
+	Subscribed   string `json:"subbed"`
+}
+
+// WsHeartBeat defines a heartbeat request
+type WsHeartBeat struct {
+	ClientNonce int64 `json:"ping"`
+}
+
+// WsDepth defines market depth websocket response
+type WsDepth struct {
+	Channel   string `json:"ch"`
+	Timestamp int64  `json:"ts"`
+	Tick      struct {
+		Bids      []interface{} `json:"bids"`
+		Asks      []interface{} `json:"asks"`
+		Timestamp int64         `json:"ts"`
+		Version   int64         `json:"version"`
+	} `json:"tick"`
+}
+
+// WsKline defines market kline websocket response
+type WsKline struct {
+	Channel   string `json:"ch"`
+	Timestamp int64  `json:"ts"`
+	Tick      struct {
+		ID     int64   `json:"id"`
+		Open   float64 `json:"open"`
+		Close  float64 `json:"close"`
+		Low    float64 `json:"low"`
+		High   float64 `json:"high"`
+		Amount float64 `json:"amount"`
+		Volume float64 `json:"vol"`
+		Count  int64   `json:"count"`
+	}
+}
+
+// WsTrade defines market trade websocket response
+type WsTrade struct {
+	Channel   string `json:"ch"`
+	Timestamp int64  `json:"ts"`
+	Tick      struct {
+		ID        int64 `json:"id"`
+		Timestamp int64 `json:"ts"`
+		Data      []struct {
+			Amount    float64 `json:"amount"`
+			Timestamp int64   `json:"ts"`
+			ID        big.Int `json:"id,number"`
```

Comment:
routines.go exchange Huobi websocket error - math/big: cannot unmarshal "\"10106462341333121613826\"" into a *big.Int

can change this to a float64, or would have to create your own type which implements its own UnmarshalJSON

Reply:
I just moved the type to huobi_types. I'll change it to float64 for now 👍